### PR TITLE
Added indexes for year, season and episode

### PR DESCRIPTION
**Pull request by SubJunk**
We should also probably have compound indexes for combinations of these, but this is a start

### CODE REVIEW

**Comment by SubJunk** [comment]
@js-kyle it seems that this line https://github.com/UniversalMediaServer/api/pull/445/files#diff-052be7cd6463de48e1dbb623e51ddae178c3301830d48b70e287e36681e0aa16R29 is causing the failure, but I'm scratching my head about that. Why would adding an index on that field cause another index to not be used?

**Comment by js-kyle** [comment]
@SubJunk From looking at this, I think there is a race condition happening with the index build, that is prevalent when you have added an additional index to the schema.

If you replace the line you've added to create an extra document with ` await new Promise(resolve => setTimeout(resolve, 1000));` it passes every time, but without that it is just flakey. So I feel like it is a non issue, but there might be some Mongoose tweaks about indexing, it could also be due to us using the mongodb in memory server

File: test/models/FailedLookups.spec.ts [hunk 1]
```diff
@@ -39,6 +39,7 @@ describe('Failed Lookups Model', () => {
   describe('Indexes', () => {
     it('should use index when find by osdbHash', async() => {
       await FailedLookupsModel.create({ osdbHash: '8e245d9679d31e12' });
+      await FailedLookupsModel.create({ osdbHash: '8e245d9679d31e13' });
```
Comment (js-kyle) [inline test/models/FailedLookups.spec.ts:42]:
```suggestion
      await FailedLookupsModel.init();
      await FailedLookupsModel.create({ osdbHash: '8e245d9679d31e12' });
```

The above Promise fufills when the index build is done (which is in the background by default) so removes the race condition from the test

**Comment by SubJunk** [comment]
Fixed, but maybe we should have more tests for it since I'm a bit confused now

**Comment by SubJunk** [comment]
Sweet thanks, keen to merge?

**Pull request merged**